### PR TITLE
RU: keras and eager

### PR DESCRIPTION
**Pull request by 0101011**
Please do not merge until Monday, more fixes and translations coming

### CODE REVIEW

**Comment by 0101011** [comment]
@mkaze Hi Masoud,

I've just tried to edit a notebook using the latest Jupyter version and it rewrote the whole backend markdown structure just as you have faced it a day before.

However, if you save a restructured file with Colab the structure goes back to the original (with all the colab-s and null-s in the markdown)

Here's the workflow:

1. Before editing the file check what markdown we've got (Jupyter or Colab): it's necessary as different notebooks saved in different editors in this repo
2. Use the appropriate editor 😉

The generated diff will be clear and easy to read.

**Comment by mkaze** [comment]
@0101011 Hi Andrew,

Actually, I edited one of the notebooks with Colab and the diff was almost OK; however, I tried to edit another notebook (again, with Colab) a few days ago and the diff was a mess! So I gave up!

Now you mention that some of them work with Jupyter editor and some of them with Colab. It's very frustrating but I will try it and see what happens. Thank you very much for this.

**Comment by MarkDaoust** [comment]
This can also help with the reviews, it ignores the json formatting and looks at the rendered result only:

https://app.reviewnb.com/tensorflow/docs/pull/252

**Comment by 0101011** [comment]
Mark, will give it a true in an hour

At least they rewrite each other 😃

**Comment by 0101011** [comment]
@haesun Hi Haesun,

Check out the issue we’ve had above:

Would be interested to hear your take and experience you’ve had while editing notebooks too 👍

**Comment by 0101011** [comment]
@rickiepark Hi Haesun,

Check out the issue we’ve had above:

Would be interested to hear your take and experience you’ve had while editing notebooks too 👍

**Comment by rickiepark** [comment]
@0101011  Hi Andrew,
I might not fully understand this thread. :)
If someone has trouble editing your notebook, I recommend plain editor like sublime text.
I think that It’s good practice when you make PR about notebooks in /en folder.
But It is a difficult approach if you have a lot of things to change. :)

**Comment by 0101011** [comment]
1. Basically, you can see the issue yourself in this commit:
https://github.com/tensorflow/docs/pull/252/commits/3e851e2c32575b0eff2225f6400b4191382aaf48

2. The all-red diff was generated by Jupyter which replaced the original json format generated by Colab: you might face the same diff as soon as you start translating `guide/keras`

But yeah, using Sublime for pure text edit might solve the problem partially, although it's not that visual comparing with Colab or Jupyter 😀

**Comment by rickiepark** [comment]
Hi Andrew,
I edited with Jupyter and tested with Colab.
I don’t understand why you diff Jupyter code with Colab code.
Is `guide/keras` different from `tutorial/keras`?

**Comment by 0101011** [comment]
1. Yup, I do the same :)

2. It IS quite different as you compare the JSON structure between [the original](https://github.com/tensorflow/docs/blob/master/site/en/guide/keras.ipynb?short_path=a3b8867) and [the Jupyter generated file](https://github.com/tensorflow/docs/pull/252/commits/2db15a1b842c651e4815f59856df99f56a2e4ac9)

Masoud's concern was it messes with generated diff (see all-red commit) and thus makes commit's changes unreadable.

Current solution is to use a third-party app to ignore json and compare text only:
https://app.reviewnb.com/tensorflow/docs/pull/252

**Comment by rickiepark** [comment]
Thank you for explanation. :)
Yes, Colab’s json format is different from Jupyter’s.
But fomatting issue is the same in /en folder.
If someone makes PR with small changes of notebook, It’s best to use plain editor.
Masoud’s PR have same problem(#258).
Alternatively, How about adding comments for editing platform(jupyter version)?
As you may know, we can’t control Colab version.
So in this case Jupyter is recommeded.

**Comment by lamberta** [comment]
Hi Andrew, let me know when you're ready with this PR and I will find a reviewer. Thanks!

**Comment by 0101011** [comment]
This one will be finished and proofread this weekend.

Yep, I translate keras.ipynb as this guide is the first page users see when they switch to Russian on Tensorflow.org

My current workflow is:

1. Open Tensorflow.org
2. Head on to Learn section
3. Pick the most important chunk of text

I'll go through sequentially top to bottom (as it goes [in the index on the left](https://www.tensorflow.org/tutorials/), next being Eager and Estimators.

If you have other suggestions for better docs to translate ASAP - send me a link 🤗

🏀 Throwing an idea: 

I see you publishing new guides weekly, maybe we can translate new ones concurrently or on the next day? (English, Korean, Spanish, Russian, Chinese etc.)

There might be a benefit of reaching more audience for those who do marketing for devs on Twitter and YouTube.

**Comment by lamberta** [comment]
Thanks, Andrew!

Confirmed everything runs:
* https://colab.research.google.com/github/0101011/docs/blob/master/site/ru/guide/eager.ipynb
* https://colab.research.google.com/github/0101011/docs/blob/master/site/ru/guide/keras.ipynb
* https://colab.research.google.com/github/0101011/docs/blob/master/site/ru/tutorials/keras/basic_classification.ipynb

@ezhulenev Eugene, when you get a chance, can you give these docs a quick look? Thank you

File: site/ru/guide/eager.ipynb [hunk 1]
```diff
@@ -0,0 +1,1743 @@
...
+    {
+      "metadata": {
+        "id": "EGjDcGxIqEfX",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "\n",
+        "\n",
+        "Eager execution в TensorFlow - это окружение императивного программирования, в котором все операции выполняются мгновенно, без построения графов: операции возвращают конкретные значения, вместо создания вычислительных графов. Это позволяет легко запускать любой код в TensorFlow, упрощает отладку моделей и устраняет необходимость в шаблонном boilerplate коде. Пройди все этапы в этом руководстве, просто запуская примеры кода в интерпретаторе `Python` ниже.\n",
```
Comment (ezhulenev) [inline site/ru/guide/eager.ipynb]: `это окружение императивного программирования, в котором все операции выполняются `
->
`это режим интерактивного выполнения всех вычислений, когда все операции выполняются мгновенно`

?

"окружение императивного программирования" совсем не понятно о чем речь идет

File: site/ru/guide/eager.ipynb [hunk 2]
```diff
@@ -0,0 +1,1743 @@
...
+        "\n",
+        "* *Интуитивный интерфейс* — структурируй свой код и используй стандартные структуры \n",
+        "данных Python. Быстро проверяй гипотезы с помощью небольших моделей на малых данных\n",
+        "* *Легкая отладка кода* — Производи любые операции непосредственно на готовых \n",
+        "моделях и проверяй изменения. Используй стандартные инструменты для отладки \n",
+        "Python кода для незамедлительного отчета об ошибках\n",
+        "* *Естественный порядок выполнения* — Используй порядок выполнения Python вместо порядка графов, тем самым упрощая спецификацию динамических моделей\n",
+        "\n",
+        "Eager execution поддерживает большиноство операций TensoFlow и ускорение при помощи GPU.\n",
+        "Смотри собрание основных примеров, которые можно\n",
```
Comment (ezhulenev) [inline site/ru/guide/eager.ipynb]: "Смотри основные примеры"

File: site/ru/guide/eager.ipynb [hunk 3]
```diff
@@ -0,0 +1,1743 @@
...
+        "\n",
+        "Eager execution является гибкой платформой машинного обучения для проведения\n",
+        "исследований и экспериментов. Ее основные преимущества:\n",
+        "\n",
+        "* *Интуитивный интерфейс* — структурируй свой код и используй стандартные структуры \n",
+        "данных Python. Быстро проверяй гипотезы с помощью небольших моделей на малых данных\n",
+        "* *Легкая отладка кода* — Производи любые операции непосредственно на готовых \n",
+        "моделях и проверяй изменения. Используй стандартные инструменты для отладки \n",
+        "Python кода для незамедлительного отчета об ошибках\n",
+        "* *Естественный порядок выполнения* — Используй порядок выполнения Python вместо порядка графов, тем самым упрощая спецификацию динамических моделей\n",
```
Comment (ezhulenev) [inline site/ru/guide/eager.ipynb]: "вместо графа вычислений" ? 

В целом "Используй порядок выполнения Python вместо порядка графов" звучит так себе но ничего хорошего в голову не приходит

File: site/ru/guide/eager.ipynb [hunk 4]
```diff
@@ -0,0 +1,1743 @@
...
+        "    super(MySimpleLayer, self).__init__()\n",
+        "    self.output_units = output_units\n",
+        "\n",
+        "  def build(self, input_shape):\n",
+        "    # Метод `build` вызывается первый раз, когда используется слой.\n",
+        "    # Создание переменных на build() позволяет тебе сделать их форму\n",
+        "    # зависимой от входной формы и таким образом устранить необходимость\n",
+        "    # пользователю уточнять формы полностью. Также возможно создавать переменные\n",
+        "    # во время __init__(), если ты уже знаешь их полные формы.\n",
+        "    self.kernel = self.add_variable(\n",
```
Comment (ezhulenev) [inline site/ru/guide/eager.ipynb]: "форма" вместо "shape" мне кажется неверно. Я бы сказал размерность.

`Создание переиенных в build() позволяет задать размерность тензоров в зависимостт от размерности входных параметров, ...`

https://ru.wikipedia.org/wiki/%D0%A2%D0%B5%D0%BD%D0%B7%D0%BE%D1%80 - dimension==размерность

File: site/ru/guide/eager.ipynb [hunk 5]
```diff
@@ -0,0 +1,1743 @@
...
+          "id": ""
+        },
+        "id": "VrfLnhNPqEf3",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "model = tf.keras.Sequential([\n",
+        "  tf.keras.layers.Dense(10, input_shape=(784,)),  # нужно указать входящую форму\n",
```
Comment (ezhulenev) [inline site/ru/guide/eager.ipynb]: указать размерность входных данных

File: site/ru/guide/eager.ipynb [hunk 6]
```diff
@@ -0,0 +1,1743 @@
...
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "a639YaF4qEf-",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Необязательно устанавливать входящую форм для классов `tf.keras.Model`,\n",
```
Comment (ezhulenev) [inline site/ru/guide/eager.ipynb]: указывать размерностьт входных данных

**Comment by 0101011** [comment]
Привет!

1. Заметки очень ценные, как будет время - пробегись ещё по keras (второй), мне тоже некоторые моменты кажутся спорными

2. Третий ноутбук в ссылке Билли - это тот же гайд по классификации, там просто пара исправлений: можно не смотреть

3. Указывай, если даже что-то звучит странно, а как исправить не знаешь - я тоже подумаю

File: site/ru/guide/eager.ipynb (hunk 1 above)
Comment (0101011) [reply site/ru/guide/eager.ipynb]: Точно! Про `режим` забыл, хотя по-русски eager - это режим и есть:
Исправляю везде.

Comment (0101011) [reply site/ru/guide/eager.ipynb]: Исправил заглавное определение так, стало гораздо чище:

`Eager execution в TensorFlow - это интерактивный режим, в котором все вычисления выполняются мгновенно. Операции в eager возвращают конкретные значения, вместо построения вычислительных графов.`

File: site/ru/guide/keras.ipynb [hunk 7]
```diff
@@ -0,0 +1,1298 @@
...
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Импортируем tf.keras\n",
+        "\n",
+        "`tf.keras` - это реализация [спецификации Keras API](https://keras.io){:.external} в TensorFlow. \n",
+        "Это высокоуровневый API для построения \n",
+        "моделей глубокого обучения с первоклассной поддержкой \n",
+        "функционала TensorFlow, например [eager execution](#eager_execution),\n",
+        "конвееры `tf.data` и алгоритмы оценки [Estimators](./estimators.md).\n",
```
[the remark this answers is not in the document]
Comment (0101011) [reply site/ru/guide/keras.ipynb]: 1. `pipiline `можно опустить, поскольку в коде Keras он не используется: если бы использовался, то обязательно нужно было бы и дать толкование, и указать где параметр в коде

2. Сейчас исправил на русский вариант `Затем определим функцию *загрузки и обработки данных*` :)

File: site/ru/guide/keras.ipynb [hunk 8]
```diff
@@ -0,0 +1,1298 @@
...
+        "id": "lj03RamP0E13"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "`tf.keras` может запускать любой совместимый с Keras код, но тем не менее помни:\n",
+        "\n",
+        "* Версия `tf.keras` в последнем релизе TensorFlow может быть не самой\n",
+        "последнией версией `keras` из индекса PyPI. Всегда проверяй версию при помощи `tf.keras.__version__`\n",
+        "* Когда ты [сохраняешь только веса модели](#weights_only), `tf.keras` по умолчанию\n",
+        "  сохраняет их в [формате контрольной точки](./checkpoints.md). Используй `save_format='h5'`\n",
```
[the remark this answers is not in the document]
Comment (0101011) [reply site/ru/guide/keras.ipynb]: Спорный момент - поищу:

1. В просторечье чекпоинты так и называем
2. Сходу отпишусь, что в разных доках именуют все `контрольными точками`, ибо уже устоялось с 90-х :) (см. русские доки Postgres, [O'Reilly "High Performance Spark"](https://books.google.ru/books?id=-AtgDwAAQBAJ&printsec=frontcover#v=onepage&q&f=false) и др.)

**Comment by 0101011** [comment]
> `Прямой проход` режет слух, но судя по хабру это достаточно широко используется, и ничего взамен я предложить тоже не могу.

Я тоже испугался, когда перевел 😀
Не только на хабре, но и в LingvoScience и переводах MatLab так и пишут - "прямой проход".

Если замену понятнее не найдем, то оставим так, потому что уже используется.

🔴 Если ты пьешь что-то горячее - поставь, потому что:

burn

**Pull request closed without merging**